moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -531,14 +532,14 @@ bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void
   {
     return false;
   }
-  double d = fcl::distance(o1, o2, fcl::DistanceRequestd(cdata->req->enable_nearest_points), fcl_result);
+  thread_local double d = fcl::distance(o1, o2, fcl::DistanceRequestd(cdata->req->enable_nearest_points), fcl_result);
```

Comment:
Same issue. It will actually **never call this distance computation method again** after the first call in each thread...